libbeat/common/file/file_rotator.go

```diff
@@ -0,0 +1,243 @@
+package file
```

Comment:
I moved the `FileRotator` over to the common/file package because it's a reusable piece of code (file output and logging). And I refactored it. Plus it should be renamed b/c it stutters.

Reply:
I'm not even sure if log rotation should be a task of the beat itself or an external log rotation tool. We should definitively revisit how it works.

Reply:
We should make it possible for beats to work with an external log rotator (`logrotate`). The beat needs to be able to accept a signal from `logrotate` that triggers the logger to write to a new file. We can make it a requirement to be `logrotate` compatible and make this happen.

Overall I like our current behavior because it just works (especially on Windows where there is no built it tool for rotating).